packages/core/src/delegation.js

```diff
@@ -21,6 +21,71 @@ export const isLink =
  */
 export const isDelegation = proof => !Link.isLink(proof)
 
+/**
+ * Takes on or more delegations and returns all delegated capabilities in
+ * UCAN 0.10 format, expanding all the special forms like `with: ucan:*` and
+ * `can: *` to an explicit forms.
+ *
+ * @template {[API.Delegation, ...API.Delegation[]]} T
+ * @param {T} delegations
+ * @returns {API.InferAllowedFromDelegations<T>}
+ */
+export const allows = (...delegations) => {
+  /** @type {API.Allows} */
+  let allow = {}
+  for (const delegation of delegations) {
+    for (const capability of delegation.capabilities) {
+      // If uri is `ucan:*` then we include own capabilities along with
+      // delegated
+      const capabilities =
+        capability.with === 'ucan:*'
+          ? iterateCapabilities(delegation)
+          : [capability]
+
+      for (const { with: uri, can, nb } of capabilities) {
+        const resource = allow[uri] || (allow[uri] = {})
+        const abilities = resource[can] || (resource[can] = [])
+        abilities.push({ ...Object(nb) })
+      }
+    }
+  }
+
+  return /** @type {API.InferAllowedFromDelegations<T>} */ (allow)
+}
+
+/**
+ * Function takes a delegation and iterates over all the capabilities expanding
+ * all the special forms like `with: ucan:*` and `can: *`.
+ *
+ * @param {API.Delegation} delegation
+ * @returns {Iterable<API.Capability>}
+ */
+const iterateCapabilities = function* ({ issuer, capabilities, proofs }) {
+  for (const { with: uri, can, nb } of capabilities) {
+    if (uri === 'ucan:*') {
+      yield {
+        with: issuer.did(),
+        can,
```

Comment:
if `can` is `*` here, it seems it would not be expanded

Reply:
Yes that is correct, however that is expected here as well. When you say `{ "with": "ucan:*", "can": "*" }` it means I re-delegate everything I own or have been delegated.

This specific yield is everything I own, which is `{ with: my.did(), can: "*" }`.

Reply:
Added code comment to clarify this